panel/layout.py

```diff
@@ -15,6 +15,26 @@
 from .viewable import Reactive, Viewable
 
 
+def no_height(layout):
+    """
+    Whether the supplied layout has no height
+    """
+    if isinstance(layout, BkBox):
+        for child in layout.children:
+            if isinstance(child, BkBox):
+                if no_height(child):
+                    return False
+
+            height = getattr(child, 'height', None)
+            if height is not None:
+                return False
+    elif isinstance(BkWidgetBox):
+        return False
+    elif getattr(child, 'height', None) is not None:
+        return False
+    return True
+
+
```

Comment:
It really hurts my brain to follow the logic of a method named `no_height`; is there any way it could be changed to `has_height`?

Reply:
Fixed.

Reply:
Much better now, and shorter!  Now I can follow the logic without any brain meltdowns. :-)